AEFeinstein/Swadge-IDF-5.0
# Implement ESPNOW in Emulator

### Description



This adds the networking code from last year, minimally updated for the new callbacks and to fix some warnings.

It also adds a bit more networking usage to the demo mode, and fixes the overall ESPNOW functionality by adding a call to check the ESPNOW queue from the main swadge loop.

And finally, this adds a python script in `/tools` which just listens for the simulated ESPNOW messages and prints them nicely.

### Test Instructions



Run two emulator instances and open Demo mode in both of them. There's now a square with counters for sent and received packets that also changes color when the connection status changes. Pressing B sends a packet, so you should see the sent counter on that emulator and the received counter on the other emulator increase.

For extra fun, you can also run `./tools/monitor_emu_wifi.py` while you do that and see if it looks correct.

### Ticket Links



Closes #62 

### Readiness Checklist
